tests/IResearch/IResearchFeature-test.cpp

```diff
@@ -1883,15 +1883,26 @@ TEST_F(IResearchFeatureTest, test_async_schedule_wait_indefinite) {
   feature.queue(arangodb::iresearch::ThreadGroup::_1, 0ms,
                 Task(deallocated, mutex, cond, count, feature));
 
-  auto const end = std::chrono::steady_clock::now() + 10s;
-  while (!count) {
-    std::this_thread::sleep_for(10ms);
-    ASSERT_LE(std::chrono::steady_clock::now(), end);
+  {
+    auto const end = std::chrono::steady_clock::now() + 10s;
+    while (!count) {
+      std::this_thread::sleep_for(10ms);
+      ASSERT_LE(std::chrono::steady_clock::now(), end);
```

Comment:
I think this is not safe.
If we go to sleep less than 10ms before `end`, the assertion can be violated.
Obviously that would also mean that it took 10 seconds for the queue to execute the task, which we can probably consider a test failure as well.
But it may be more understandable to write the loop as:
```
while (!count && std::chrono::steady_clock::now() < end) {
  std::this_thread::sleep_for(10ms);
}
```